netket/graph/lattice.py

```diff
@@ -246,6 +315,114 @@ def atoms_coord(self):
         """
         return self._atoms_coord
 
+    @property
+    def coords(self):
+        """
+        Returns list of coordinates of lattice points
+        """
+
+        return self._coords
+
+    def translation_perm(self):
+        perms = []
+        for vec in self.basis_vectors:
+            perm = []
+            for coord in self._coords:
+                hash_coord = coord.copy() + vec
+                hash_coord = _np.matmul(hash_coord, self._inv_dims) % 1
+                # make sure 1 and 0 are treated the same
+                hash_coord = _np.around(hash_coord * 10 ** tol_digits).astype(int) % (
+                    10 ** tol_digits
+                )
+                hash_coord = HashableArray(hash_coord)
+                perm.append(self._hash_positions[hash_coord])
+
+            perms.append(tuple(perm))
+        return tuple(perms)
+
+    def rotation_perm(self, period, axes=[0, 1]):
+        perm = []
+        axes = list(axes)
+        angle = 2 * pi / period
+        rot_mat = _np.array(
+            [[_np.cos(angle), -_np.sin(angle)], [_np.sin(angle), _np.cos(angle)]]
+        )
+
+        rot_coords = self._coords.copy()
+        rot_coords[:, axes] = _np.matmul(rot_coords[:, axes], rot_mat)
+
+        for hash_coord in rot_coords:
+            hash_coord = _np.matmul(hash_coord, self._inv_dims) % 1
+            hash_coord = _np.around(hash_coord * 10 ** tol_digits).astype(int) % (
+                10 ** tol_digits
+            )
+            hash_coord = HashableArray(hash_coord)
+            if hash_coord in self._hash_positions:
+                perm.append(self._hash_positions[hash_coord])
```

Comment:
You are using this hashing trick in a few places.
I think all this logic should go in it's own internal function (not method) , documented and used throughout.
So that if we decide to change the hasing logic it's easier to do as it's all in one place.
And it avoids duplciation.
and it can be tested stand-alone.

Reply:
I would also remove all the references to "hashing". That's an implementation detail. The important thing is that we have a mapping from (rounded) position to lattice site index.